landoapi/hg.py

```diff
@@ -127,19 +131,33 @@ def __init__(self, path, config=None):
     def _config_to_list(self):
         return ["{}={}".format(k, v) for k, v in self.config.items() if v is not None]
 
+    def __call__(self, request_user):
+        self.request_user = request_user
+        return self
```

Comment:
For what it's worth, this could lead to the following situation:
```
with repo("user_a"):
    # user_a is active

with repo:
    # user_a is active, again, despite not being set.
```

This mutable behaviour could accidentally be depended on in the future, which would be bad.
Fortunately, it can be resolved by having `__call__` return a disposable context manager:
```
lil_ctx_manager = repo("user_a")
with lil_ctx_manager:
    # user_a is set

# or, how this would be used:
with repo("user_a")
    # user_a is set

with repo:
    # doesn't have a user set, as expected, because one isn't provided.
```

This adds some more code though, so the tradeoff is up to you :)

Reply:
Good point -- I think the existing implementation with `__enter__` and `__exit__` may be too complex to maintain with this approach, so I refactored it into two separate context managers.